test/api/v3/integration/groups/POST-groups_invite.test.js

```diff
@@ -0,0 +1,198 @@
+  });
+
+  describe('email invites', () => {
+    it('invites a user to a group by email', async () => {
+      let invite = await user.post(`/groups/${group._id}/invite`, {
+        emails: [{name: 'test', email: 'test@habitca.com'}],
+      });
+
+      expect(invite).to.exist;
+    });
+
+    it('invites multiple users to a group by email', async () => {
+      let invite = await user.post(`/groups/${group._id}/invite`, {
+        emails: [{name: 'test', email: 'test@habitca.com'}, {name: 'test2', email: 'test2@habitca.com'}],
+      });
+
+      expect(invite).to.exist;
+    });
+
+    it('returns an error when invite is missing an email', async () => {
+      await expect(user.post(`/groups/${group._id}/invite`, {
+        emails: [{name: 'test'}],
+      }))
+      .to.eventually.be.rejected.and.eql({
+        code: 400,
+        error: 'BadRequest',
+        message: t('inviteMissingEmail'),
+      });
+    });
+  });
+
+  describe('user and email invites', () => {
+    it('returns an error when emails and uuids are not provided', async () => {
+      await expect(user.post(`/groups/${group._id}/invite`))
+      .to.eventually.be.rejected.and.eql({
+        code: 400,
+        error: 'BadRequest',
+        message: t('canOnlyInviteEmailUuid'),
+      });
+    });
+
+    it('invites users to a group by uuid and email', async () => {
+      let newUser = await generateUser();
+      let invite = await user.post(`/groups/${group._id}/invite`, {
+        uuids: [newUser._id],
+        emails: [{name: 'test', email: 'test@habitca.com'}],
+      });
+      let invitedUser = await newUser.get('/user');
+
+      expect(invite).to.exist;
+      expect(invitedUser.invitations.guilds[0].id).to.equal(group._id);
+    });
+  });
+
+  describe('guild invites', () => {
+    it('returns an error when invited user is already in the group', async () => {
+      let newUser = await generateUser();
+      let invite = await user.post(`/groups/${group._id}/invite`, {
+        uuids: [newUser._id],
+      });
+      let joinResult = await newUser.post(`/groups/${group._id}/join`);
+
+      expect(invite).to.exist;
+      expect(joinResult).to.exist;
+      await expect(user.post(`/groups/${group._id}/invite`, {
+        uuids: [newUser._id],
+      }))
+      .to.eventually.be.rejected.and.eql({
+        code: 401,
+        error: 'NotAuthorized',
+        message: t('userAlreadyInGroup'),
+      });
+    });
+
+    it('returns an error when invited user is already invited to the group', async () => {
+      let newUser = await generateUser();
+      let invite = await user.post(`/groups/${group._id}/invite`, {
+        uuids: [newUser._id],
+      });
+
+      expect(invite).to.exist;
+      await expect(user.post(`/groups/${group._id}/invite`, {
+        uuids: [newUser._id],
+      }))
+      .to.eventually.be.rejected.and.eql({
+        code: 401,
+        error: 'NotAuthorized',
+        message: t('userAlreadyInvitedToGroup'),
+      });
+    });
```

Comment:
It should just throw an error as shown in a test above. I'll make an explicit test for multiple users. Throwing an error in the middle may not be the best though, because, some users will have invites. Hmm..